bluesky/bundlers.py

```diff
@@ -176,12 +168,15 @@ async def _prepare_stream(self, desc_key, objs_read):
                 'run_uid': self._run_start_uid,
                 'data_keys': data_keys.keys()}
         )
-        self._descriptors[desc_key] = (objs_read, descriptor_doc)
-        return descriptor_doc
+        self._descriptor_objs[desc_key] = objs_read
+        return self._descriptors[desc_key].descriptor_doc, self._descriptors[desc_key].compose_event, objs_read
 
-    async def _ensure_cached(self, obj):
+    async def _ensure_cached(self, obj, collect=False):
         if obj not in self._describe_cache:
-            await self._cache_describe(obj)
+            if collect:
+                await self._cache_describe_collect(obj)
```

Comment:
We need to think about this one a bit more, `_cache_describe_collect` so `_ensure_cached` called twice will do describe collect twice

Reply:
Also, `_prepare_stream` is called which takes data out of `_describe_cache` so I don't know how that works...